### PR TITLE
Moving pieces part 2

### PR DESCRIPTION
**Pull request by SijiaDavis**
Moving pieces - Part 2: Remove "middle page" (pieces#show) and replace with jquery

There's still one Rubocop complaint that we are not sure about how to fix. 
Rubocop suggests that we replace the class variable `@@shared_connection` with an instance variable `@share_connection`, but tests will fail if we make that change.

Thank you in advance for any suggestions on how to fix this complaint.

### CODE REVIEW

File: spec/support/share_db_connection.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class ActiveRecord::Base
```
Comment (mscoutermarsh) [inline spec/support/share_db_connection.rb]: exclude this file from rubocop

http://rubocop.readthedocs.io/en/latest/configuration/#includingexcluding-files

File: spec/features/move_chess_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    it 'updates the coordinates of the piece', js: true do
+      piece = FactoryGirl.create(:piece)
+      game = piece.game
+
+      visit game_path(game)
+
+      piece_to_drag = page.find("#piece-#{piece.id}")
+      target = page.find('td[data-x="0"][data-y="4"]')
+
+      piece_to_drag.drag_to(target)
```
Comment (mscoutermarsh) [inline spec/features/move_chess_spec.rb]: wow - nice

File: app/views/games/show.html.erb [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
-  <% end %>
+<table width='800' height='800' align='center' border='2' class='chess-board'>
+
+    <% (0..7).each do |row| %>
+        <tr>
+            <% (0..7).each do |column| %>
+                <% if @game.render_piece(column, 7 - row) != nil %>
+                    <td width='100' height='100' class='<%= column.even? ^ row.even? ? "white" : "black" %> droppable' data-x='<%= column %>' data-y='<%= 7 - row %>'>
+                        <% piece = @game.retrieve_piece(column, 7 - row) %>
+                        <%= image_tag(@game.render_piece(column, 7 - row), class: "draggable", id: "piece-#{piece.id}", data: {x: column, y: 7-row} ) %>
```
Comment (mscoutermarsh) [inline app/views/games/show.html.erb]: nice use of `data`

File: app/models/piece.rb [hunk 4]
```diff
@@ -44,11 +43,13 @@ def obstructed?(destination_x, destination_y)
     if x_distance.zero?
       # the two locations are on the same vertical line
       chess_in_between = game.pieces.where(x_coordinate: x_coordinate)
-                             .where(y_coordinate: y_coordinate + 1...destination_y)
+                             .where(y_coordinate: y_coordinate + 1...destination_y).pluck(:id)
```
Comment (mscoutermarsh) [inline app/models/piece.rb]: pluck = 👍

File: app/controllers/pieces_controller.rb [hunk 5]
```diff
@@ -6,10 +6,10 @@ def show
   end
 
   def update
-    @piece = Piece.find(params[:id])
-    @piece.update_attributes(piece_params)
-    @game = @piece.game
-    render template: 'games/show'
+    piece = Piece.find(params[:id])
+    return unless piece.update(piece_params)
```
Comment (mscoutermarsh) [inline app/controllers/pieces_controller.rb]: I'd do something like

```
if piece.update(piece_params)
  @game = piece.game
  head :ok, location: @game
end

head :bad_request
```

File: spec/support/share_db_connection.rb (hunk 1 above)
Comment (mumtaz7) [inline spec/support/share_db_connection.rb]: ah thank you! and thank you for all your feedback, mike :)

**Pull request merged**